### ISSUE TITLE
Changing post parameters before send

### ISSUE THREAD

**Issue by aymanfarhat** [vitalets/x-editable#305]
Hello everyone I am having an issue with x-editable's default post parameters namely name, pk and value. 

Here is the thing, on the server I am getting the value of name for the form via $_POST["name"] but name in my context doesnt mean a field name but a name representing something. I don't want to get its value via $_POST["value"] within that script for several reasons. Is there a way to rename the values before sending them to the server?

In other words all I need to do is give the name field the new value before send. I don't care about the value and pk fields.

Thanks.

**Comment by tarim** [comment]
I tried data-params={}. And, it works good for single word. If you add more than one word or has space, it will doesn't work. any suggestion?

**Comment by abelvf30** [comment]
I'm using X-Editable and I want to pass a simple value to other page using the _post method and I receive this error:

Notice: Undefined index: first_name in ...

Here my code:

this is a simple html page that have this code:
<form method="post" action="update.php">
    <a href="#" id="first_name" data-type="text" data-url="/post" data-title="Edit First Name" class="editable editable-click">BILL GATES</a>
</form>

Here my javascript code:
$(document).ready(function() {

//toggle `popup` / `inline` mode
$.fn.editable.defaults.mode = 'popup';     

//--------------------------------------------------------------------------------------------
//First Name
$('#first_name').editable({         
    validate: function(value) {
   //Not Empty  
       if($.trim(value) == '') return 'The First Name is required';

       //More than 3 and less than 20
       if($.trim(value.length) <4) return 'The First Name must be more than 3 and less than 20 characters';
       //Less than 20
       if($.trim(value.length) >20) return 'The First Name must be more than 3 and less than 20 characters';           
    }
});

And here I want to capture the value in the php code:
$first_name = $_POST['first_name'];
echo $first_name."<br>";

and I receive this: 


Notice: Undefined index: first_name in ...

Please any advice or suggestion?

Regards...
